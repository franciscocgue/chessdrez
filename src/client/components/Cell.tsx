import React from 'react'

import styles from './Cell.module.css';

interface Props {
    posV: 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8,
    posH: 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8,
    piece: null | 'peon' | 'alfil' | 'caballo' | 'torre' | 'reina' | 'rey'
}

const Cell = ({ posV, posH, piece }: Props) => {

    const teamColor = (posH % 2 && posV % 2) || (!(posH % 2) && !(posV % 2)) ? 'negras' : 'blancas';

    return (
        <div className={styles[teamColor]}>

        </div>
    )
};

export default Cell;