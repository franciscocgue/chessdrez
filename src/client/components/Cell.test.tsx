import React from 'react'
import renderer from 'react-test-renderer';
import Cell from './Cell';
import peonwhite from '../assets/peon_blanco.png';

it('shows possible moves when pawn cell 78 hovered', () => {
    const component = renderer.create(
        <Cell
            key={'78'}
            img={peonwhite}
            row={7}
            col={8}
        />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})