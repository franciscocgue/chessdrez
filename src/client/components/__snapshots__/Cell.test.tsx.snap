// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows possible moves when pawn cell 78 hovered 1`] = `
<div
  className="white cell "
  draggable={true}
  onDragEnter={[Function]}
  onDragStart={[Function]}
  onDrop={[Function]}
  onMouseEnter={[Function]}
>
  <img
    className="draggable icon"
    src="test-file-stub"
    width="95%"
  />
</div>
`;
